regex.go

```diff
@@ -171,8 +193,13 @@ func CreateRegex(stringBufferInBytes uint32) Regex {
 	// by GC, this finalizer ensures that regexes are being used as efficiently as possible by maximizing pool rotations.
 	// Hopefully, this would be caught during development and not in production.
 	runtime.SetFinalizer(pr, func(pr *privateRegex) {
-		if pr.mod != nil && ShouldPanic {
-			panic("Finalizer found a Regex that was never closed")
+		if pr.mod != nil {
+			if leakHandler := getRegexLeakHandler(); leakHandler != nil {
+				leakHandler()
+			} else if ShouldPanic {
+				panic("Finalizer found a Regex that was never closed")
+			}
+			_ = pr.Close()
```

Comment:
I don't know of any reason this wouldn't be safe to do at finalization time, but Daylon should give his opinion here too.

Reply:
Looking through the code a bit, I can't find nor think of a good reason as to why this shouldn't be fine, so I approve.